sql/logictest: TestLogic failed due to mismatched crdb_internal table IDs

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2352947&tab=buildLog) on [master@9bfd0673de1255c52fd18fcafa710145f2f4bd3a](https://github.com/cockroachdb/cockroach/commits/9bfd0673de1255c52fd18fcafa710145f2f4bd3a):

```
                        test    52   [62]         62   myschema_t2       65
                        test    52   [62]         62   myschema_seq1     66
                        test    52   [62]         62   myschema_t3       67
                        test    52   otherschema  70   otherschema_v1    71
                        test    52   otherschema  70   otherschema_t1    72
                        test    52   otherschema  70   otherschema_seq1  73
                        test    52   [75]         75   scdrop1_t1        78
                        test    52   [75]         75   scdrop1_t2        79
                        test    52   [76]         76   scdrop2_t1        80
                        test    52   [76]         76   scdrop2_v1        81
                        test    52   [77]         77   scdrop3_v1        82
                        test    52   privs        90   tbl               93
                        test    52   privs        90   usage_tbl         96
                        new_db  103  [104]        104  bar               105
                        new_db  103  public       29   public_table      107
                        new_db  103  testuser     106  test_table        108
                        new_db  103  public       29   test_table        109
                        
                    but found (query options: "rowsort" -> ignore the following ordering of rows) :
                        test    52   myschema2    54   tb                55
                        test    52   myschema2    54   v                 58
                        test    52   myschema2    54   s                 59
                        test    52   myschema2    54   tb2               60
                        test    52   [62]         62   myschema_t1       64
                        test    52   [62]         62   myschema_t2       65
                        test    52   [62]         62   myschema_seq1     66
                        test    52   [62]         62   myschema_t3       67
                        test    52   otherschema  70   otherschema_v1    71
                        test    52   otherschema  70   otherschema_t1    72
                        test    52   otherschema  70   otherschema_seq1  73
                        test    52   [75]         75   scdrop1_t1        78
                        test    52   [75]         75   scdrop1_t2        79
                        test    52   [76]         76   scdrop2_t1        80
                        test    52   [76]         76   scdrop2_v1        81
                        test    52   [77]         77   scdrop3_v1        82
                        test    52   privs        90   tbl               93
                        test    52   privs        90   usage_tbl         96
                        new_db  103  [104]        104  bar               105
                        new_db  103  public       29   public_table      108
                        new_db  103  testuser     107  test_table        109
                        new_db  103  public       29   test_table        110
                logic.go:1670: 
                     testdata/logic_test/schema:567: too many errors encountered, skipping the rest of the input
=== RUN   TestLogic/fakedist-vec-auto-disk/schema
=== PAUSE TestLogic/fakedist-vec-auto-disk/schema
=== CONT  TestLogic/fakedist-vec-auto-disk/schema
        --- FAIL: TestLogic/fakedist-vec-auto-disk/schema (7.81s)
            logic.go:2153: let $schema_id = 62
=== RUN   TestLogic/fakedist-vec-auto-disk
    --- FAIL: TestLogic/fakedist-vec-auto-disk (0.45s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #54948 sql/logictest: TestLogic failed cluster setting updated but timed out waiting to read new value  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.2)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2358495&tab=buildLog) on [master@8285d8a9daeb3a3fa692c34eb6f4d6093a65e85c](https://github.com/cockroachdb/cockroach/commits/8285d8a9daeb3a3fa692c34eb6f4d6093a65e85c):

```
                        test    52   myschema2    54   tb2               60
                        test    52   [62]         62   myschema_t1       64
                        test    52   [62]         62   myschema_t2       65
                        test    52   [62]         62   myschema_seq1     66
                        test    52   [62]         62   myschema_t3       67
                        test    52   otherschema  70   otherschema_v1    71
                        test    52   otherschema  70   otherschema_t1    72
                        test    52   otherschema  70   otherschema_seq1  73
                        test    52   [75]         75   scdrop1_t1        78
                        test    52   [75]         75   scdrop1_t2        79
                        test    52   [76]         76   scdrop2_t1        80
                        test    52   [76]         76   scdrop2_v1        81
                        test    52   [77]         77   scdrop3_v1        82
                        test    52   privs        90   tbl               93
                        test    52   privs        90   usage_tbl         96
                        new_db  103  [104]        104  bar               105
                        new_db  103  public       29   public_table      107
                        new_db  103  testuser     106  test_table        108
                        new_db  103  public       29   test_table        109
                        
                    but found (query options: "rowsort" -> ignore the following ordering of rows) :
                        test    52   myschema2    54   tb                55
                        test    52   myschema2    54   v                 58
                        test    52   myschema2    54   s                 59
                        test    52   myschema2    54   tb2               60
                        test    52   [62]         62   myschema_t1       64
                        test    52   [62]         62   myschema_t2       65
                        test    52   [62]         62   myschema_seq1     66
                        test    52   [62]         62   myschema_t3       67
                        test    52   otherschema  70   otherschema_v1    71
                        test    52   otherschema  70   otherschema_t1    72
                        test    52   otherschema  70   otherschema_seq1  73
                        test    52   [75]         75   scdrop1_t1        78
                        test    52   [75]         75   scdrop1_t2        79
                        test    52   [76]         76   scdrop2_t1        80
                        test    52   [76]         76   scdrop2_v1        81
                        test    52   [77]         77   scdrop3_v1        82
                        test    52   privs        90   tbl               93
                        test    52   privs        90   usage_tbl         96
                        new_db  103  [104]        104  bar               105
                        new_db  103  public       29   public_table      108
                        new_db  103  testuser     107  test_table        109
                        new_db  103  public       29   test_table        110
                logic.go:1669: 
                     testdata/logic_test/schema:567: too many errors encountered, skipping the rest of the input
=== RUN   TestLogic/fakedist-vec-auto/schema
=== PAUSE TestLogic/fakedist-vec-auto/schema
=== CONT  TestLogic/fakedist-vec-auto/schema
        --- FAIL: TestLogic/fakedist-vec-auto/schema (6.85s)
            logic.go:2152: let $schema_id = 62

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #54948 sql/logictest: TestLogic failed cluster setting updated but timed out waiting to read new value  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.2)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/testdata/logic_test/schema
+++ b/pkg/sql/logictest/testdata/logic_test/schema
@@ -45,6 +45,16 @@ CREATE TYPE myschema.typ AS ENUM ('user', 'defined', 'schema');
 CREATE VIEW myschema.v AS SELECT x FROM myschema.tb;
 CREATE SEQUENCE myschema.s
 
+query TITITI rowsort
+SELECT
+  database_name, parent_id, schema_name, parent_schema_id, name, table_id
+FROM crdb_internal.tables
+WHERE database_name = 'test'
+----
+test  52  myschema  54  tb  55
+test  52  myschema  54  v   58
+test  52  myschema  54  s   59
+
 query I
 SELECT * FROM myschema.tb
 ----
@@ -533,35 +543,6 @@ query I colnames
 SELECT * FROM testuser.test_table
 ----
 a
-
-query TITITI rowsort
-SELECT
-  database_name, parent_id, schema_name, parent_schema_id, name, table_id
-FROM crdb_internal.tables
-WHERE database_name IN ('test', 'new_db')
-----
-test    52   myschema2    54   tb                55
-test    52   myschema2    54   v                 58
-test    52   myschema2    54   s                 59
-test    52   myschema2    54   tb2               60
-test    52   [62]         62   myschema_t1       64
-test    52   [62]         62   myschema_t2       65
-test    52   [62]         62   myschema_seq1     66
-test    52   [62]         62   myschema_t3       67
-test    52   otherschema  70   otherschema_v1    71
-test    52   otherschema  70   otherschema_t1    72
-test    52   otherschema  70   otherschema_seq1  73
-test    52   [75]         75   scdrop1_t1        78
-test    52   [75]         75   scdrop1_t2        79
-test    52   [76]         76   scdrop2_t1        80
-test    52   [76]         76   scdrop2_v1        81
-test    52   [77]         77   scdrop3_v1        82
-test    52   privs        90   tbl               93
-test    52   privs        90   usage_tbl         96
-new_db  103  [104]        104  bar               105
-new_db  103  public       29   public_table      107
-new_db  103  testuser     106  test_table        108
-new_db  103  public       29   test_table        109
 
 # Ensure that when we create a schema, it inherits privileges from its parent
 # database, but only those which are valid for schemas.

